### DISCUSSION TITLE
Lock Column once Set

### DISCUSSION THREAD

**Discussion by connorwinston** [directus/directus#15526] [Archive]
Is there a way to do the following for a SINGLE role?

1. Allow a Column to be Null
2. Allow Not Required to be Set on Created
3. If Set, Deny any updated to that column

We have a collection of members who can add a profile picture for identity purposes, however, one they set a picture we want to deny the ability to update it so it forces them to contact us.

I have conditions set up to where if it is set then make the field read only, however, this doesn't work in Bath Editing or API access.

Or is there a way to disable batch updates to certain fields in the admin app?

**Comment by rijkvanzanten** [comment]
> We have a collection of members who can add a profile picture for identity purposes, however, one they set a picture we want to deny the ability to update it so it forces them to contact us.

Do you want them to be able to set a profile picture once after their main "profile" already exists, or only on the first creation of the whole profile? If the latter, it should be sufficient to set the create permissions, but leave the update permissions. If the former, you could set that up with a flow. By using a filter trigger you can deny the update on the fly based on whatever step you chose. In this case, you could fetch the current value for the picture of the current profile that's being updated, and use a validation operation to prevent the update when the profile picture is already set 🙂

**Reply by connorwinston** [reply]
Ok, that was what I was thinking but wanted to make sure there wasn't a way with a permissions to do that. We will go with the flows way.
